Gordon360/Services/NewsService.cs

```diff
@@ -153,9 +184,161 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
-            return (StudentNewsViewModel)newsItem;
+            return newsItem;
+        }
+
+        /// <summary>
+        /// Takes a filepath for an image, navigates to it, collects the raw data
+        /// of the file and converts it to base64 format. 
+        /// 
+        /// The base64 data will not include the first part of a base64 image ("data:image/...").
+        /// This is because this part is removed in every image before being submitted, and it
+        /// is readded in the frontend before being displayed.
+        /// 
+        /// This helper function does not perform any error checking; every place that calls it
+        /// checks that the path is not empty. Theoretically if it isn't empty it's certainly a valid path.
+        /// </summary>
+        /// <param name="imagePath">The path to the image</param>
+        /// <returns>The base64 content of the image</returns>
+        private string GetBase64ImageDataFromPath(string imagePath)
+        {
+            using (System.Drawing.Image image = System.Drawing.Image.FromFile(imagePath))
+            using (MemoryStream data = new MemoryStream())
+            {
+                image.Save(data, image.RawFormat);
+                byte[] imageBytes = data.ToArray();
+                string base64Data = System.Convert.ToBase64String(imageBytes);
+                return base64Data;
+            }
+        }
+
+        /// <summary>
+        /// Iterates through a collection of news items that have just been queried from
+        /// the database, replacing the Image entry of each (which contains the path to an image)
+        /// with the image itself. This is done because when the frontend makes requests for news items,
+        /// it needs the image data and not the path, but we store the path in the DB for efficiency and
+        /// storage reasons.
+        /// </summary>
+        /// <param name="items">The collection of student news entries</param>
+        /// <returns>The collection of entries with base64 data in their image variable</returns>
+        private IEnumerable<StudentNewsViewModel> InsertImageData(IEnumerable<StudentNewsViewModel> items)
+        {
+            foreach (StudentNewsViewModel item in items)
+            {
+                string imagePath = item.Image;
+
+                if (imagePath != null)
+                {
+                    item.Image = GetBase64ImageDataFromPath(imagePath);
+                }   
+            }
+            return items;
+        }
+
+        /// <summary>
+        /// Uploads a news image
+        /// Can be used to add an image to a new posting or to replace an image
+        /// for an existing posting.
+        /// 
+        /// Because editPosting calls this method to replace whatever is currently stored
+        /// (if anything) with new data, and because the new data can actually include
+        /// an image property of null (if the user removed the image from their post), it's
+        /// possible for 
```

Comment:
This comment ends prematurely ("it's possbile for" what?). Also, The summary of a method should not comment on when/where/how the method is used elsewhere in the code - both because that is bound to change and because it is not the job of a comment. The summary is meant to explain what the method does and how it should be used. The method doesn't care who uses and where (or it shouldn't - and if it does that indicates that it's too tightly coupled).